src/wallet/api.rs

```diff
@@ -1209,8 +1201,16 @@ impl Wallet {
     /// - Change output costs
     /// - Privacy considerations
     ///
+    /// Implements mandatory address grouping with strategic selection:
+    /// 1. Groups UTXOs by address
+    /// 2. Tries single group optimization (finds smallest group that covers target+fees)
+    /// 3. Falls back to multi-group selection with privacy-focused sorting
+    /// 4. Uses all-or-nothing principle (always selects entire groups)
+    /// 5. Runs normal coin selection on remaining UTXOs if needed
+    ///
```

Comment:
Shorten this doc to explaining just the grouped utxo behavior. 

"Always prefers to spend reused addresses first to preserve privacy. Selects more utxos if total reused addrs amount isn't adequate".

Then remove these detailed comments before the main group selection loop, and explain whats happening there.